src/Symfony/Bridge/Twig/Extension/CodeExtension.php

```diff
@@ -25,17 +25,19 @@ class CodeExtension extends AbstractExtension
     private $fileLinkFormat;
     private $rootDir;
     private $charset;
+    private $projectDir;
 
     /**
      * @param string|FileLinkFormatter $fileLinkFormat The format for links to source files
      * @param string                   $rootDir        The project root directory
      * @param string                   $charset        The charset
      */
-    public function __construct($fileLinkFormat, string $rootDir, string $charset)
+    public function __construct($fileLinkFormat, string $rootDir, string $charset, string $projectDir = null)
     {
         $this->fileLinkFormat = $fileLinkFormat ?: ini_get('xdebug.file_link_format') ?: get_cfg_var('xdebug.file_link_format');
         $this->rootDir = str_replace('/', DIRECTORY_SEPARATOR, dirname($rootDir)).DIRECTORY_SEPARATOR;
         $this->charset = $charset;
+        $this->projectDir = $projectDir;
```

Comment:
AFAIK in practice `projectDir` will equal `rootDir` here, i think we should keep relying on this path (rootDir).

Basically we do the same thing here:

https://github.com/symfony/symfony/blob/bcd8058478f5c5ba3f9813aedff663a81be19551/src/Symfony/Bridge/Twig/Extension/CodeExtension.php#L179-L181

Reply:
`$rootDir` works by convention and should not be used anymore.